src/main/java/se/sundsvall/supportmanagement/api/validation/impl/ValidRoleConstraintValidator.java

```diff
@@ -0,0 +1,33 @@
+package se.sundsvall.supportmanagement.api.validation.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import se.sundsvall.supportmanagement.api.model.metadata.Role;
+import se.sundsvall.supportmanagement.api.validation.ValidRole;
+import se.sundsvall.supportmanagement.integration.db.model.enums.EntityType;
+import se.sundsvall.supportmanagement.service.MetadataService;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.util.List;
+
+import static java.util.Collections.emptyList;
+import static java.util.Optional.ofNullable;
+
+public class ValidRoleConstraintValidator extends AbstractTagConstraintValidator implements ConstraintValidator<ValidRole, String> {
+	@Autowired
+	private MetadataService metadataService;
+
+	@Override
+	public boolean isValid(final String value, final ConstraintValidatorContext context) {
+		final var namespace = getPathVariable(PATHVARIABLE_NAMESPACE);
+		final var municipalityId = getPathVariable(PATHVARIABLE_MUNICIPALITY_ID);
+		return  !metadataService.isValidated(namespace, municipalityId, EntityType.ROLE) ||
+			isValid(value, getRoles(namespace, municipalityId), context);
+	}
+
+	private List<String> getRoles(String namespace, String municipalityId) {
```

Comment:
I'm just curious:

The roleRepository has a existsByNamespaceAndMunicipalityIdAndName().
Wouldn't it be better to call this method (through the service-layer) instad of looping through a list of roles?

Is there any advantages with the current (looping) solution?

Reply:
I can not see any advantages with current solution and I can not see any reason for not changing handling of all types as you propose.

Reply:
When I look closer I find that the disadvantage is that the errorMessage will not show valid tags in your proposal.

Reply:
ok